### PR TITLE
CI: adding email notifications of CI success/failure

### PR DESCRIPTION
**Pull request by alyssawilk**
having azure add github comments emailing on precheck failure, CI failure, or CI success.

Risk Level: low risk of email spam if the script does not WAI
Testing: manual
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

**Comment by alyssawilk** [comment]
tested format fail and CI success.  I can also test CI fail or we can just land and iterate :-)

**Comment by phlax** [comment]
im very -1 to this i think, and im not clear what problem it solves

my email inbox is already swamped by mails from this repo making it increasingly hard to separate signal from noise

nevertheless, i do use email to prioritise which notifications to give attention to. The net effect of this is that i would be more likely to unsubscribe from most PRs or perhaps block email notifications altogether

imho ci failures in precheck are only of concern to the person pushing the pr, and the maintainer that is reviewing, and for the latter only *when* they are reviewing

for postsubmit failures the bigger issues is flakes, and i think the on-rotation maintainer should probably be monitoring these to check that flakes are actually flakes

in short i dont think it serves any purpose to email everyone subscribed to a PR that there has been a ci failure when most PRs will fail many times before landing

**Comment by phlax** [comment]
on that note - im already swamped by often useless meta emails from repokitteh

**Comment by daixiang0** [comment]
I think it would be better to close as default, users can do something to open it if like it. 

I have mark repokitteh's mail as junk since it is usually useless, I hope mail from repokitteh can be closed if possible.

**Comment by alyssawilk** [comment]
So there's definitely a need for push notifications for CI - folks having to poll github to know if they failed format checks is a PITA.

I agree it'd be nice to email the person who the PR belonged.  I have azure do that for me, but you can't set up those emails without admin rights to the repo, so non-maintainers can't do that.  I'd suggested having a general mailing list so folks who aren't admins can sub to the list and filter out emails, but @htuch vetoed that, and suggested we should have what is implemented here. I'd be happy to revisit and do that instead, if @envoyproxy/maintainers prefer.

I like CI success emails because frequently when I approve a PR I want to merge, but can't because CI is not finished.  And again it's back to polling github.  I also know some reviewers won't review until CI is green so I thought it was useful for that.
Totally open to other ideas if you have them

**Comment by phlax** [comment]
can we add some tag notation for doing this

when i first saw `/wait` and `waiting` i assumed that was what it did

i think this defo needs to be opt in and preferably on a case-by-case basis

**Comment by alyssawilk** [comment]
how would one opt in? I think this is especially useful for newcomers to the project who don't know to check CI so I'm not sure that'd be a win.

**Comment by phlax** [comment]
> how would one opt in? 

im guessing with a tag - ie im a reviewer, i want to know when ci completes and what is the result

> I think this is especially useful for newcomers to the project who don't know to check CI so I'm not sure that'd be a win.

i think most people who create a PR know that github displays ci results at the bottom of the pull request

im really not convinced that having a noisybot helps anyone
